menpo/shape/mesh/base.py

```diff
@@ -236,6 +309,257 @@ def n_tris(self):
+                    target_name = target_mesh.path.stem
+                else:
+                    target_name = 'Target'
+                figure_name = 'Heatmap between {} and {}'.format(source_name,
+                                                                 target_name)
+            else:
+                figure_name = None
+        else:
+            figure_name = figure_id
+
+        diff = (source_mesh.points.astype(np.float32)-target_mesh.points.astype(np.float32))**2
+        distances_between_meshes = np.sqrt(diff.sum(axis=1))
+        scaled_distances_between_meshes = distances_between_meshes*scale_value
+
+        if inline:
+            try:
+                from menpo3d.visualize import HeatmapInlineViewer3d
+                renderer = HeatmapInlineViewer3d(figure_name, new_figure,
+                                                 self.points, self.trilist,
+                                                 self.landmarks)
+                render_return = renderer._render(scaled_distances_between_meshes,
+                                                 type_cmap, scalar_range,
+                                                 show_statistics)
+
+                if render_return is not renderer:
+                    renderer.close()
+                    return
+                return renderer
+            except ImportError as e:
+                from menpo.visualize import Menpo3dMissingError
+                raise Menpo3dMissingError(e)
+        else:
+            try:
+                from menpo3d.visualize import HeatmapViewer3d
+                renderer = HeatmapViewer3d(figure_name, new_figure,
+                                           self.points, self.trilist)
+
+                if type_cmap == 'hot_r':
+                    type_cmap = 'hot'
+                renderer.render(scaled_distances_between_meshes,
+                                type_cmap, scalar_range, show_statistics)
+
+                return renderer
+            except ImportError as e:
+                from menpo.visualize import Menpo3dMissingError
+                raise Menpo3dMissingError(e)
+
+    def decimate(self, percentage_reduction=0.75,
+                 type_reduction='quadric', **kwargs):
+        """
+        Decimate the mesh specifying the percentage (0,1) of triangles to
+        be removed
+
+        Parameters
+        ----------
+        percentage_reduction: float (default: 0.75)
+                   The percentage of triangles to be removed.
+                   It should be in (0, 1)
+
+        type_reduction : str (default: quadric)
+                         The type of decimation as:
+                             'quadric' : Quadric decimation
+                             'progressive : Progressive decimation
+        Returns
+        -------
+        mesh : :map:`TriMesh`
+            A new mesh that has been decimated.
+        """
+        try:
+            from menpo3d.vtkutils import decimate_mesh
```

Comment:
It's cool to have this method but new PR and we can discuss if we want to expose it like this or if it should stay in menpo3d?

Reply:
Yup, sure. I was not pretty sure how to deal with this kind of addition and which one is the proper and more pythonic way  to add a new method without adding vtk dependency. We can discuss this in a new PR.